oidcauth/http.go

```diff
@@ -0,0 +1,98 @@
+package oidcauth
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/coreos/go-oidc"
+	"github.com/skuid/spec/middlewares"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// User is an OIDC authenticated user
+type User struct {
+	Email string `json:"email"`
+}
+
+// key is an unexported type for keys defined in this package.
+// This prevents collisions with keys defined in other packages.
+type key int
+
+// userKey is the key for user.User values in Contexts. It is
+// unexported; clients use user.NewContext and user.FromContext
+// instead of using this key directly.
+var userKey key
+
+// NewContext returns a new Context that carries value u.
+func NewContext(ctx context.Context, u *User) context.Context {
+	return context.WithValue(ctx, userKey, u)
+}
+
+// FromContext returns the User value stored in ctx, if any.
+func FromContext(ctx context.Context) (*User, bool) {
+	u, ok := ctx.Value(userKey).(*User)
+	return u, ok
```

Comment:
nitpick: you can just return them directly, right?

Reply:
Actually, not for a type assertion like this 
https://play.golang.org/p/_CiThDck8T

Reply:
Interesting, I forgot this was a "comma ok" operation.